masakari-controller/controller/masakari_config.py

```diff
@@ -27,10 +27,96 @@
 import socket
 import threading
 import errno
+import uuid
+
 from oslo_config import cfg
 from oslo_log import log as logging
+from oslo_context import context
+
 
 LOG = logging.getLogger(__name__)
+_request_store = threading.local()
+context._request_store = _request_store
+
+
+def generate_request_id():
+    """Generate a unique request id."""
+    return 'req-%s' % uuid.uuid4()
+
+
+class RequestContext(context.RequestContext):
+
+    """Helper class to represent useful information about a request context.
+
+    Stores information about the security context under which the user
+    accesses the system, as well as additional request information.
+    """
+
+    user_idt_format = '{user} {tenant} {domain} {user_domain} {p_domain} ' \
+        + '{thread_name} {record_identifier}'
+
+    def __init__(self, auth_token=None, user=None, tenant=None, domain=None,
+                 user_domain=None, project_domain=None, is_admin=False,
+                 read_only=False, show_deleted=False, request_id=None,
+                 resource_uuid=None, overwrite=True, roles=None,
+                 thread_name=None, record_identifier=None):
+        """Initialize the RequestContext
+
+        :param overwrite: Set to False to ensure that the greenthread local
+                          copy of the index is not overwritten.
+        """
+        self.thread_name = thread_name
+        self.record_identifier = record_identifier
```

Comment:
record_identifier is the value that is able to specify masakari db record uniquely.
record_identifier value is not always UUID.
So I didn't set record_identifier values to resource_uuid.

But, I decided to fix my code to include record_identifier values in thread name.
So I remove record_identifier from instance variables.